plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1935,12 +1932,7 @@ private void writeCheckpointIfNeeded(ConnectorSession session, SchemaTableName t
             checkpointWriterManager.writeCheckpoint(session, snapshot);
         }
         catch (Exception e) {
-            if (ignoreCheckpointWriteFailures) {
-                LOG.warn(e, "Failed to write checkpoint for table %s for version %s", table, newVersion);
-            }
-            else {
-                throw new TrinoException(GENERIC_INTERNAL_ERROR, format("Failed to write checkpoint for table %s for version %s", table, newVersion), e);
-            }
+            LOG.error(e, "Failed to write checkpoint for table %s for version %s", table, newVersion);
```

Comment:
Let's capture the reasoning from our conversation, why we don't fail here

Reply:
Comment added